Xamarin.PropertyEditing.Mac/Controls/Custom/PropertyButton.cs

```diff
@@ -0,0 +1,74 @@
+﻿using System;
+using System.Windows.Input;
+using AppKit;
+using CoreGraphics;
+
+namespace Xamarin.PropertyEditing.Mac
+{
+	public class PropertyButton : NSButton
+	{
+		NSMenu popUpContextMenu;
+		public NSMenu PopUpContextMenu {
+			get { return popUpContextMenu; }
+			set { popUpContextMenu = value; }
+		}
+
+		CommandMenuItem reset;
+
+		public PropertyButton ()
+		{
+			AlternateImage = NSImage.ImageNamed ("property-button-default-mac-active-10");
+			Cell = new NSButtonCell {
+				HighlightsBy = 1,
+			};
+			Bordered = false;
+			Enabled = true;
+			Image = NSImage.ImageNamed ("property-button-default-mac-10");
+			ImageScaling = NSImageScale.AxesIndependently;
+			Title = string.Empty;
+			TranslatesAutoresizingMaskIntoConstraints = false;
+
+			Activated += (sender, e) => { 
+				var menuOrigin = this.Superview.ConvertPointToView (new CGPoint (Frame.Location.X - 1, Frame.Location.Y + Frame.Size.Height + 4), null);
+
+				var popupMenuEvent = NSEvent.MouseEvent (NSEventType.LeftMouseDown, menuOrigin, (NSEventModifierMask)0x100, 0, this.Window.WindowNumber, this.Window.GraphicsContext, 0, 1, 1);
+
+				NSMenu.PopUpContextMenu (popUpContextMenu, popupMenuEvent, this);
+			};
+
+			popUpContextMenu = new NSMenu ();
```

Comment:
We could wait to construct this until we actually are opening the menu, instead of constructing one for 100 properties that won't open it.

Reply:
Ok

Reply:
hmm I may need to talk to you about doing the late creation of this.